### PR TITLE
fix(ui): add context-notice component and refresh token stats after every chat turn, solve #45230, #45034, #45192, #45794

### LINKED ISSUE #45034: Control UI: Token stats not showing in real-time chat
**描述**
在 Control UI 对话页面，底部的 token 统计（↑输入 ↓输出 R缓存 %ctx占用）只在页面刷新后才显示，实时消息完成后不会自动出现。

**复现步骤**
1. 打开 Control UI 对话页面
2. 发送一条消息
3. 助手回复后，底部没有统计显示
4. 刷新页面后统计出现

**环境**
- OpenClaw: 2026.3.12
- 平台: WSL2
- 浏览器: Chrome/Edge

**预期行为**
消息完成后应立即显示 token 统计。

### LINKED ISSUE #45794: [Bug]: Control UI chat pane can break at 100% context used even when context-notice CSS is already fixed
### Bug type

Behavior bug (incorrect output/state without crash)

### Summary

On macOS with OpenClaw `2026.3.12`, the Control UI can still render a broken/blank chat area when a session reaches `100% context used`, even if the known `.context-notice` CSS regression is already patched locally.

This appears to be more than the oversized warning icon bug.

In this case:

- the warning pill itself looked normal after a local CSS hotfix
- the chat area still broke when the session hit `100% context used` (user report: `206.4k / 200k`)
- session data continued to append normally on disk

So the failure looks like a Control UI render/state bug on the context exhaustion boundary, not session corruption.

### Steps to reproduce

1. Run OpenClaw `2026.3.12` on macOS.
2. Open Control UI in Chrome.
3. Use a long-running main session until the UI reports `100% context used`.
4. In the same period, let model fallback/compaction activity happen normally.
5. Observe that the chat pane can render incorrectly or appear blank, even though the session still exists and continues to append messages.

### Expected behavior

- Context exhaustion should show a bounded warning.
- The chat history should remain visible.
- Input should either stay usable or be explicitly disabled with a clear action such as "Compact Session".
- If compaction fails, the UI should show a normal error state instead of collapsing the chat area.

### Actual behavior

- The chat pane becomes visually broken/blank at the `100% context used` boundary.
- In this repro, the context warning pill itself was already normal due to a local CSS hotfix, so the failure was not just the oversized icon overlay.
- Session data was still intact and continued updating on disk.

### OpenClaw version

2026.3.12

### Operating system

macOS

### Install method

npm global

### Model

Primary: `anthropic/claude-opus-4-6`

Fallback seen during repro: `gemini/gemini-3.1-pro-preview`

### Provider / routing chain

local gateway loopback

### Config file / key location

`~/.openclaw/openclaw.json`

### Additional provider/model setup details

This machine is already using a local Control UI hotfix bundle because the installed `2026.3.12` bundle still lacks `.context-notice` CSS selectors.

Override in use:

`gateway.controlUi.root = /Users/lucifinil_chen/.openclaw/runtime-overrides/control-ui-hotfix-20260314/dist/control-ui`

Because the issue still reproduces with that override active, this looks like a second bug beyond the known CSS regression.

### Logs, screenshots, and evidence

Relevant gateway log lines during the same repro window:

```text
2026-03-14T14:07:34.497+08:00 [compaction] Full summarization failed, trying partial: Summarization failed: 429 {"error":{"message":"API 密钥额度已用完，请联系管理员充值","type":"quota_exceeded"},"type":"error"}
2026-03-14T14:07:47.434+08:00 [compaction] Partial summarization also failed: Summarization failed: 429 {"error":{"message":"API 密钥额度已用完，请联系管理员充值","type":"quota_exceeded"},"type":"error"}
2026-03-14T14:07:58.289+08:00 [compaction] Full summarization failed, trying partial: Summarization failed: 429 {"error":{"message":"API 密钥额度已用完，请联系管理员充值","type":"quota_exceeded"},"type":"error"}
2026-03-14T14:08:08.940+08:00 [compaction] Partial summarization also failed: Summarization failed: 429 {"error":{"message":"API 密钥额度已用完，请联系管理员充值","type":"quota_exceeded"},"type":"error"}
```

Session-store evidence that data remained intact:

- session file: `agents/main/sessions/d0e51a49-1ea3-426a-b4dd-4416aa4018c5.jsonl`
- compaction record exists at `2026-03-14T06:08:08Z`
- a new user message exists at `2026-03-14T06:09:06Z`
- a new assistant reply exists at `2026-03-14T06:09:34Z`

That makes this look like a UI/render issue rather than session loss.

Related issues:

- #44906
- #45230
- #45513

This report is specifically about the chat pane still breaking even when the `.context-notice` CSS path has already been patched locally.

### Impact and severity

High for long-running sessions in Control UI. The conversation appears broken at the exact moment the user most needs recovery/compaction controls.

### Additional information

The active runtime is also using `memory-lancedb-pro@1.1.0-beta.8`, but this issue does not appear to be related to the memory plugin. The failure is reproducible in the Control UI/render path.

### PR DESCRIPTION
**Pull request by haoruilee**
## Summary

- **Problem 1 (#45230):** On page refresh the Control UI showed a full-screen context warning with impossible values (`550.4k / 204.8k`, usage > window). Two root causes: (a) the component used `inputTokens` (cumulative all-time input, can far exceed the context window) instead of `totalTokens` (current turn's context window usage); (b) the CSS for `.context-notice` was missing entirely, so the unstyled `<div>` expanded to fill the screen.
- **Problem 2 (#45034):** Token stats in the chat tab never updated in real-time after a message. `loadSessions()` was only called after a terminal chat event when the run ID was in `refreshSessionsAfterChat` — which only happened for `/new`/`reset` commands, never for normal turns.
- **What changed:** (1) Added `renderContextNotice` to `ui/src/ui/views/chat.ts` using `totalTokens` as `used`, with an inline SVG warning icon; (2) added `.context-notice` CSS block to `ui/src/styles/chat/layout.css` as a non-blocking inline banner with progressive amber→red coloring; (3) changed `handleTerminalChatEvent` in `ui/src/ui/app-gateway.ts` to unconditionally call `loadSessions()` after every `final` chat event.
- **What did NOT change:** No gateway logic, no session storage, no token accounting, no model/provider code touched.

## Change Type (select all)

- [x] Bug fix

## Scope (select all touched areas)

- [x] UI / DX

## Linked Issue/PR

- Closes #45034
- Closes #45230
- Closes #45192
- Closes #45794

## User-visible / Behavior Changes

- The chat tab now shows a compact amber→red inline warning banner when the active session's context usage is ≥85% of the context window. The notice shows `X% context used` and `used / limit` token counts.
- Token stats (totalTokens, contextTokens) in the session data are now refreshed after every completed chat turn, so the context notice appears and updates in real-time without a page reload.

## Security Impact (required)

- New permissions/capabilities? No
- Secrets/tokens handling changed? No
- New/changed network calls? No — `sessions.list` was already called post-chat for /new commands; it is now called for all final events (same endpoint, same parameters)
- Command/tool execution surface changed? No
- Data access scope changed? No

### Environment

- OS: Linux (WSL2 repro environment)
- Runtime/container: Node 22
- Model/provider: any

### Steps

1. Open the Control UI chat tab with a long-running session (>85% context)
2. Send a normal message (not /new)
3. After the reply: the context banner should appear immediately
4. Refresh the page: banner should show correct values (not usage > window)

### Expected

- Context notice appears as an inline compact banner after each reply
- Values show `totalTokens / contextTokens` where `totalTokens <= contextTokens`

### Actual (before fix)

- Issue #45034: no banner appeared after normal replies; only appeared after page refresh
- Issue #45230: after refresh, banner used cumulative `inputTokens` which could be 2–3× larger than `contextTokens`, showing `550.4k / 204.8k`; missing CSS caused full-screen overlay

## Evidence

- All 812 test files pass (`pnpm test`)
- `pnpm check` passes (oxlint + format)

## Human Verification (required)

- Verified scenarios: lint passes, all tests pass, TypeScript types checked
- Edge cases checked: session with no totalTokens/contextTokens → notice returns nothing (no render); ratio < 0.85 → nothing; ratio > 1.0 → clamped to 100%
- Thanks @martinfrancois checked the fix
- What you did not verify: only tested related issue

## Compatibility / Migration

- Backward compatible? Yes
- Config/env changes? No
- Migration needed? No

## Failure Recovery (if this breaks)

- How to disable/revert this change quickly: revert commit `9d6599072`
- Files/config to restore: `ui/src/ui/views/chat.ts`, `ui/src/ui/app-gateway.ts`, `ui/src/styles/chat/layout.css`
- Known bad symptoms: extra `sessions.list` call after each chat turn (very cheap read, no side effects)

## Risks and Mitigations

- Risk: `loadSessions()` now fires after every final chat event instead of only after /new. This adds one lightweight gateway request per turn.
  - Mitigation: `loadSessions` already has a guard (`if (state.sessionsLoading) return`) to prevent concurrent calls. The request is cheap (just reads the in-memory session store).

### CODE REVIEW

**Comment by martinfrancois** [comment]
I also have the same issue (#45230) and can confirm this does fix the issue.

File: ui/src/ui/app-gateway.ts [hunk 1]
```diff
@@ -291,11 +291,14 @@ function handleTerminalChatEvent(
...
-      });
-    }
+  }
+  // Refresh session list after every final event so totalTokens/contextTokens
+  // stay current in real-time, updating the context-usage notice without
+  // requiring a page reload or a /new command (fixes #45034, #45230, #45192).
+  // No overrides: respect whatever activeMinutes/limit filters the user has set
+  // in the Sessions UI rather than silently narrowing to a hard-coded 120-min window.
+  if (state === "final") {
+    void loadSessions(host as unknown as OpenClawApp);
```
[the remark this answers is not in the document]
Comment (haoruilee) [reply ui/src/ui/app-gateway.ts]: Valid observation, but not a regression we introduced and not the right fix for this PR. The `sessionsLoading` guard is a deliberate "skip if busy" pattern used identically throughout the entire codebase, `refreshChat`, `patchSession`, `deleteSessionAndRefresh` all behave the same way. A "pending refresh" flag would be a separate improvement. Our change is strictly better than before (was never called after normal turns at all).

File: ui/src/ui/views/chat.ts [hunk 2]
```diff
@@ -255,9 +255,19 @@ function renderContextNotice(
...
+  if (!limit) {
+    return nothing;
+  }
+
+  // Prefer totalTokens (per-turn snapshot) when marked fresh — it reflects
+  // the actual current context size, not the cumulative sum across all turns.
+  // When totalTokensFresh===false the snapshot is stale/missing; fall back to
+  // inputTokens capped at the context window so the ratio never exceeds 1.
+  const isFresh = session?.totalTokensFresh !== false;
+  const used = isFresh ? (session?.totalTokens ?? 0) : Math.min(session?.inputTokens ?? 0, limit);
```
[the remark this answers is not in the document]
Comment (haoruilee) [reply ui/src/ui/views/chat.ts]: Valid bug, now fixed

**Comment by BunsDev** [comment]
Closing as superseded by #71297 for the Control UI chat context-warning/session-refresh portion. #71297 resolves the current version of the same stale context notice failure with a smaller scoped patch: live session metadata is applied immediately from gateway events, overlapping session reloads are coalesced, stale totals are hidden, the context notice layout is covered by current styles, and the compact recommendation CTA is included.

This PR also contains broad unrelated provider/config/test churn, so it should not remain the canonical path for the Control UI chat fix. Any still-needed non-UI changes should be split into separate PRs.
